### ISSUE TITLE
ProjectScaffold Error on Linux Generating Documentation

### ISSUE THREAD

**Issue by Adron** [tpetricek/FSharp.Formatting#323]
Here's where I've run into the issue and made the original post: http://stackoverflow.com/questions/30992501/projectscaffold-error-on-linux-generating-documentation

I was able to run the document build based on Tomas' recommendations with this command:

    fsharpi -d:HELP docs/tools/generate.fsx and fsharpi -d:REFERENCE docs/tools/generate.fsx

...and got the following error:

    Processing the file 'docpage' failed
    Source written to: '/tmp/tmp4a1e2cc4.tmp.cs'
    Compilation errors:
     - error: (0, 0) Assembly `FakeLib, Version=3.35.2.0, Culture=neutral, PublicKeyToken=null' references `FSharp.Core, Version=4.3.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' which has a higher version number than imported assembly `FSharp.Core, Version=4.3.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'

    System.Exception: Generating HTML failed.
      at FSharp.Literate.RazorRender.handleCompile[String] (System.String source, Microsoft.FSharp.Core.FSharpFunc`2 f) [0x00000] in <filename unknown>:0 
      at FSharp.Literate.RazorRender.ProcessFileModel[IDictionary`2] (System.Type modelType, System.Object model, Microsoft.FSharp.Core.FSharpOption`1 properties) [0x00000] in <filename unknown>:0 
      at FSharp.Literate.Templating.generateFile (Microsoft.FSharp.Core.FSharpOption`1 references, System.String contentTag, IEnumerable`1 parameters, Microsoft.FSharp.Core.FSharpOption`1 templateOpt, System.String output, IEnumerable`1 layoutRoots) [0x00000] in <filename unknown>:0 
      at FSharp.Literate.Templating.processFile (Microsoft.FSharp.Core.FSharpOption`1 references, FSharp.Literate.LiterateDocument doc, System.String output, FSharp.Literate.ProcessingContext ctx) [0x00000] in <filename unknown>:0 
      at FSharp.Literate.Literate.ProcessScriptFile (System.String input, Microsoft.FSharp.Core.FSharpOption`1 templateFile, Microsoft.FSharp.Core.FSharpOption`1 output, Microsoft.FSharp.Core.FSharpOption`1 format, Microsoft.FSharp.Core.FSharpOption`1 formatAgent, Microsoft.FSharp.Core.FSharpOption`1 prefix, Microsoft.FSharp.Core.FSharpOption`1 compilerOptions, Microsoft.FSharp.Core.FSharpOption`1 lineNumbers, Microsoft.FSharp.Core.FSharpOption`1 references, Microsoft.FSharp.Core.FSharpOption`1 fsiEvaluator, Microsoft.FSharp.Core.FSharpOption`1 replacements, Microsoft.FSharp.Core.FSharpOption`1 includeSource, Microsoft.FSharp.Core.FSharpOption`1 layoutRoots, Microsoft.FSharp.Core.FSharpOption`1 generateAnchors, Microsoft.FSharp.Core.FSharpOption`1 assemblyReferences, Microsoft.FSharp.Core.FSharpOption`1 customizeDocument) [0x00000] in <filename unknown>:0 
      at <StartupCode$FSharp-Literate>.$Main+processScriptFile@184.Invoke (System.String file, System.String output) [0x00000] in <filename unknown>:0 
      at Microsoft.FSharp.Core.FSharpFunc`2[System.String,System.String].InvokeFast[Unit] (Microsoft.FSharp.Core.FSharpFunc`2 func, System.String arg1, System.String arg2) [0x00000] in <filename unknown>:0 
      at <StartupCode$FSharp-Literate>.$Main.processDirectory@199 (Microsoft.FSharp.Core.FSharpOption`1 format, Boolean processRecursive, Microsoft.FSharp.Core.FSharpFunc`2 processScriptFile, Microsoft.FSharp.Core.FSharpFunc`2 processMarkdown, System.String indir, System.String outdir) [0x00000] in <filename unknown>:0 
      at FSharp.Literate.Literate.ProcessDirectory (System.String inputDirectory, Microsoft.FSharp.Core.FSharpOption`1 templateFile, Microsoft.FSharp.Core.FSharpOption`1 outputDirectory, Microsoft.FSharp.Core.FSharpOption`1 format, Microsoft.FSharp.Core.FSharpOption`1 formatAgent, Microsoft.FSharp.Core.FSharpOption`1 prefix, Microsoft.FSharp.Core.FSharpOption`1 compilerOptions, Microsoft.FSharp.Core.FSharpOption`1 lineNumbers, Microsoft.FSharp.Core.FSharpOption`1 references, Microsoft.FSharp.Core.FSharpOption`1 fsiEvaluator, Microsoft.FSharp.Core.FSharpOption`1 replacements, Microsoft.FSharp.Core.FSharpOption`1 includeSource, Microsoft.FSharp.Core.FSharpOption`1 layoutRoots, Microsoft.FSharp.Core.FSharpOption`1 generateAnchors, Microsoft.FSharp.Core.FSharpOption`1 assemblyReferences, Microsoft.FSharp.Core.FSharpOption`1 processRecursive, Microsoft.FSharp.Core.FSharpOption`1 customizeDocument) [0x00000] in <filename unknown>:0 
      at FSI_0003.buildDocumentation () [0x00000] in <filename unknown>:0 
      at <StartupCode$FSI_0003>.$FSI_0003.main@ () [0x00000] in <filename unknown>:0 
      at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
      at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0 
    Stopped due to error

**Comment by tpetricek** [comment]
cc @matthid - have you ever seen this sort of error?

**Comment by Adron** [comment]
Another question I have regarding this, is how do I just disable the build of the docs during a full or all build anyway? Can I just take it out of some file or such?

Cheers,

**Comment by matthid** [comment]
Yes I have seen this error, and actually this was one of the reasons for the workaround removed with https://github.com/tpetricek/FSharp.Formatting/commit/87c12844eee20969998f30bae9dfc1c9512ffbdb (especially `(getItem "FSharp.Core").Replace("4.3.0.0", "4.3.1.0")`). 

Do you use an older mono version? Can you try to update mono or add this workaround if you need to be compatible?

**Comment by tpetricek** [comment]
@Adron You can comment out the line that generates the docs (or API reference) at the end of your `generate.fsx` script: https://github.com/fsprojects/ProjectScaffold/blob/master/docs/tools/generate.template#L145

**Comment by Adron** [comment]
@tpetricek @matthid I'll give your suggestions a try and play around with the docs gen in the build and out of the build.

@matthid - I should have the latest version of mono, but will check again to insure that I do. I'll get a list of the exact version I have and post it for reference.

**Comment by Adron** [comment]
Ok, I gave both suggestions a try. I did have the latest mono installed @matthid. I also tried the generate.fsx that @tpetricek suggested. That worked by just removing it, but it wasn't ideal since I'd like to have everything just working right.

So here is were things got kind of interesting. I put everything back the way it was in the generate.fsx file and did a build. It broke, as I'd suspect with the error above. Then, by happenstance I noticed I needed to upgrade a number of things on Ubuntu 14.04 itself. So I clicked install and started downloading some 600+ MB of patches and updates. That finished, I had to reboot and the fs couldn't be mounted. So I went through a reboot, came into safe mode and let Ubuntu run a package rebuild for everything installed on the operating system. When that finished, I booted back up normally and launched Visual Studio Code and the bash to get back to work resolving this issue.

That's when I did a build and everything passed perfectly. Moral of the story, it appears that Ubuntu 14.04 needs the latest patches as well as insuring the latest mono is installed.  It seems there must have been some mono dependency that was necessary for docs generation. So, I'm going to mark this as closed, and self-post a solution to Stack Overflow. I REALLY appreciate the help from both of you, cheers!

**Comment by tpetricek** [comment]
Huh, that sounds pretty mysterious. Oh well, computers :-). Glad you got it to work in the end!

**Comment by jacqueline-homan** [comment]
@Adron , @tpetricek : I am having problems with getting this error `System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0 
Stopped due to error` too. I tried applying Tomas's fix `fsharpi -d:HELP docs/tools/generate.fsx and fsharpi -d:REFERENCE docs/tools/generate.fsx`, but got a different error:

![screenshot from 2015-07-23 00 18 23](https://cloud.githubusercontent.com/assets/3537889/8843028/775998ae-30d0-11e5-9c26-e234dd74d21d.png)



 So far, I have not able to get this fixed.

**Comment by tpetricek** [comment]
@jacqueline-homan Do you have a more detailed log for the first error you were getting?

As for the second one, is the file `docs/tools/generate.fsx` actually in your current folder? (Either it isn't there, or something is going wrong when F# searches for the file...)

**Comment by jacqueline-homan** [comment]
@tpetricek , Actually I was trying to work through a tutorial on Maps where you had to send the code to F# interactive. So the first error occurred there in the F# interactive in Monodevelop and I don't have a detailed log for it. When I was researching to try to find the solution for fixing it, I came across this thread and followed what @Adron did, but then got the error I got in my screencap above. I ran the command `cat locate docs/tools/generate.fsx` in my terminal and no results came back. So that means it's not anywhere in my machine.
